web/react/components/audio_video_preview.jsx

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import Constants from '../utils/constants.jsx';
+import FileInfoPreview from './file_info_preview.jsx';
+import * as Utils from '../utils/utils.jsx';
+
+export default class AudioVideoPreview extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.handleFileInfoChanged = this.handleFileInfoChanged.bind(this);
+        this.handleLoadError = this.handleLoadError.bind(this);
+
+        this.stop = this.stop.bind(this);
+
+        this.state = {
+            canPlay: true
+        };
+    }
+
+    componentWillMount() {
+        this.handleFileInfoChanged(this.props.fileInfo);
+    }
+
+    componentDidMount() {
+        if (this.refs.source) {
+            $(ReactDOM.findDOMNode(this.refs.source)).one('error', this.handleLoadError);
+        }
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.fileUrl !== nextProps.fileUrl) {
+            this.handleFileInfoChanged(nextProps.fileInfo);
+        }
+    }
+
+    handleFileInfoChanged(fileInfo) {
+        let video = ReactDOM.findDOMNode(this.refs.video);
+        if (!video) {
+            video = document.createElement('video');
+        }
+
+        const canPlayType = video.canPlayType(fileInfo.mime_type);
+
+        this.setState({
+            canPlay: canPlayType === 'probably' || canPlayType === 'maybe'
```

Comment:
Yeah, I'm 0/5 it's not a big deal. I just prefer constants whenever there's a literal that won't change so that whoever is working on that file knows it's a meant to be a constant and not just some random literals we're using